contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,860 @@
+        emit Withdrawal(WETH, address(0), _amount);
+    }
+
+
+    /**
+     * @dev Collect the AERO token from the gauge 
+     */
+    function _collectRewardTokens() internal override {
+        // TODO do other stuff
+        super._collectRewardTokens();
+    }
+
+    /**
+     * @dev Retuns bool indicating whether asset is supported by strategy
+     * @param _asset Address of the asset
+     */
+    function supportsAsset(address _asset) public view override returns (bool) {
+        return _asset == WETH;
+    }
+
+    /**
+     * @dev Internal method to respond to the addition of new asset / pTokens
+            We need to give the Aerodrome pool approval to transfer the
+            asset.
+     * @param _asset Address of the asset to approve
+     * @param _pToken Address of the pToken
+     */
+    // solhint-disable-next-line no-unused-vars
+    function _abstractSetPToken(address _asset, address _pToken)
+        internal
+        override
+    {
+        IERC20(_asset).safeApprove(address(positionManager), type(uint256).max);
+        IERC20(_asset).safeApprove(address(swapRouter), type(uint256).max);
+    }
+
+    /**
+     * @dev Approve the spending of all assets 
+     */
+    function safeApproveAllTokens()
+        external
+        override
+        onlyGovernor
+        nonReentrant
+    {
+        IERC20(WETH).safeApprove(address(positionManager), type(uint256).max);
+        IERC20(OETHb).safeApprove(address(positionManager), type(uint256).max);
+        IERC20(WETH).safeApprove(address(swapRouter), type(uint256).max);
+        IERC20(OETHb).safeApprove(address(swapRouter), type(uint256).max);
+    }
+
+    /***************************************
+            Balances and Fees
+    ****************************************/
+
+    /**
+     * @dev Get the total asset value held in the platform
+     * @param _asset      Address of the asset
+     * @return balance    Total value of the asset in the platform
+     */
+    function checkBalance(address _asset)
+        external
+        view
+        override
+        returns (uint256)
+    {
+        return underlyingAssets;
+    }
+
+    /**
+     * @dev Returns the balance of both tokens in a given position (excluding fees)
+     * @return amountWETH Amount of WETH in position
+     * @return amountOETHb Amount of OETHb in position
+     */
+    function getPositionPrincipal()
+        public
+        view
+        returns (uint256 amountWETH, uint256 amountOETHb)
+    {
+        if (tokenId == 0) {
+            return (0,0);
+        }
+
+        uint160 sqrtRatioX96 = getPoolX96Price();
+        (amountWETH, amountOETHb) = helper.principal(
+            positionManager,
+            tokenId,
+            sqrtRatioX96
+        );
+    }
+
+    function getPoolX96Price()
+        public
+        view
+        returns (uint160 sqrtRatioX96) {
+            (sqrtRatioX96, , , , ,) = clPool.slot0();
+        }
+
+    function _getPositionInfo() 
```

Comment:
⚪ We never use the tokensOwed0 and tokensOwed1 from this, just the liquidity information. Would make our code cleaner if we don't return these values that are never used. Perhaps we could call this _getLiquidity.

Reply:
Adding to this, this is a view function. But doesn't have the `view` keyword

Reply:
thanks, both comments addressed here: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/ecbbfef460e23bcc8df4d6d3b8df9a8eab51dd67